### PR TITLE
CodeAction as real CodeAction. Example and Test.

### PR DESCRIPTION
**Pull request by dhuebner**
Signed-off-by: Dennis Huebner <dennis.huebner@gmail.com>

### CODE REVIEW

**Comment by cdietrich** [comment]
@dhuebner am on holiday => somebody else needs to review

**Comment by dhuebner** [comment]
@cdietrich Thanks, have a nice holiday!

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.xtend [hunk 1]
```diff
@@ -40,14 +41,14 @@ class CodeActionService implements ICodeActionService {
...
-				case UNSORTED_MEMBERS: commands += d.fixUnsortedMembers(document, resource, params)
+				case INVALID_NAME: actions += Either.forLeft(d.fixInvalidName(document, resource, params))
+				case UNSORTED_MEMBERS: actions += Either.forRight(d.fixUnsortedMembers(document, resource, params))
 			}
 		}
-		return commands.map[Either.forLeft(it)];
+		return actions
 	}
 
 	def private Command fixInvalidName(Diagnostic d, Document doc, XtextResource res, CodeActionParams params) {
```
Comment (svenefftinge) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.xtend]: Is there a reason you translated only one of these two methods to use code actions?

Comment (dhuebner) [reply org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.xtend]: Yes, to keep both tested

**Pull request merged**